openedx/core/djangoapps/user_api/accounts/tests/test_views.py

```diff
@@ -479,6 +481,17 @@ def test_get_account_by_user_id_non_integer(self, non_integer_id):
         response = client.get(url + f'?lms_user_id={non_integer_id}')
         assert response.status_code == status.HTTP_400_BAD_REQUEST
 
+    @mock.patch('openedx.core.djangoapps.user_api.accounts.views.is_email_retired')
+    def test_get_retired_user_from_email(self, mock_is_email_retired):
+        """
+        Tests that the retired user from email cannot be accessed and shows an error message.
+        """
+        mock_is_email_retired.return_value = True
+        client = self.login_client('staff_client', 'staff_user')
+        url = reverse("accounts_detail_api")
+        response = client.get(url + f'?email={quote(self.user.email)}')
```

Comment:
why is the quoting needed?

Reply:
yes, the test email here contains `+` so we need to escape them before they get parsed otherwise `+` will be replaced by spaces.